backend/routes/api/spots.js

```diff
@@ -42,4 +42,30 @@ router.post('/', requireAuth, validateSpot, async (req,res) => {
     return res.json(newSpot);
 });
 
+router.delete('/:id', requireAuth, (req, res) => {
+    const spotId = req.params.id
+    const userId = req.user.id;
+
+    //Locate the spot 
+    const spotIndex = spots.findIndex(spot => spot.id === spotId);
```

Comment:
What is the "findIndex" method and could you provide the link to the documentation where you found it from?

Reply:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/findIndex

Reply:
Cool, so the findIndex has to be called on an array since it's a built in array method. What is the "spots" that you're called the method on?

Reply:
So i am not sure of what you are asking. So i am going to respond with this - const spotIndex = spots.findIndex(spot => spot.id === spotId).      --> this should be looking for index in the 'spot' of the array where the spot id should be matching the provided ID.  if the spot index is a -1, then that means the spot isn't existing in the array return an error.

Reply:
Right, but where are you getting "spots" from? In order find the index of where a spot id matches the provided id, you would need the array to search.

Reply:
But after further thinking. I am going to change it, something is telling me that wouldn't work correctly
. So stand by for the updated code

Reply:
I have updated the code and changed it. Need it to be revieweed